spackbot/pr_run_pipeline.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import logging
+import random
+import requests
+import re
+import os
+
+from .helpers import found, alias_regex
+from gidgethub import routing
+
+
+logger = logging.getLogger(__name__)
+router = routing.Router()
+
+# We can only make the request with a GITLAB TOKEN
+GITLAB_TOKEN = os.environ.get("GITLAB_TOKEN")
+
+
+async def run_pipeline(event, gh):
+    """
+    Make a request to re-run a pipeline.
+    """
+    # Get the pull request number
+    number = event.data["issue"]["pull_request"]["url"].split("/")[-1]
+
+    # We need the pull request branch
+    response = requests.get(event.data["issue"]["pull_request"]["url"])
+    pr = response.json()
+
+    # Get the sender of the PR - do they have write?
+    sender = event.data["sender"]["login"]
+    repository = event.data["repository"]
+    collaborators_url = repository["collaborators_url"]
+
+    # If they don't have write, we don't allow the command
+    if not await found(gh.getitem(collaborators_url, {"collaborator": sender})):
+        logger.info(f"Not found: {sender}")
+        return (
+            "Sorry %s, I cannot do that for you. Only users with write can make this request!"
+            % sender
+        )
+
+    # We need the branch name plus number to assemble the GitLab CI
+    branch = pr["head"]["ref"]
+    branch = "github/pr%s_%s" % (number, branch)
+
+    # Spack has project ID 2
+    url = "https://gitlab.spack.io/api/v4/projects/2/pipeline?ref=%s" % branch
```

Comment:
can you make this part a global constant like:

```python
gitlab_spack_project_url = "https://gitlab.spack.io/api/v4/projects/2"
```

Reply:
sure.